File: chapter10/typescript/USCoffeeMachine.ts
import { USAppliance } from "./USAppliance";


/**
 * USCoffeeMachine implements run method which make sure input voltage can run US Coffee Machine
 *
 * @Author: Bridget
 */
export class USCoffeeMachine extends USAppliance {

    /**
     * check voltage input is 110 and can run US Coffee Machine to Brew Coffee
     *
     * @param volt
     */
     public run(volt: number): void{
         let isVoltCompatible = this.checkVoltageCompatible(volt);
         console.log("USCoffeeMachine: Input Voltage is ", volt);
         if (!isVoltCompatible) {
            console.log("Voltage - (" + volt + ") is not supported to use US Coffee Machine");
         } else {
            console.log("Brew Coffee!");
         }
     }
}

File: chapter10/java/designpattern/src/main/java/com/designpattern/USCoffeeMachine.java
package com.designpattern;

/**
 * USCoffeeMachine implements run method which make sure input voltage can run US Coffee Machine
 *
 * @Author: Bridget
 */

public class USCoffeeMachine implements USAppliance {
    /**
     * check input voltage is 110 and can run US Coffee Machine to Brew Coffee
     *
     * @param volt
     */
    @Override
    public void run(int volt) {
        boolean isVoltCompatible = checkVoltageCompatible(volt);
        System.out.println("USCoffeeMachine: Input Voltage is " + volt);
        if (!isVoltCompatible) {
            System.out.println("Voltage (" + volt + ") is not supported to use US Coffee Machine");
        } else {
            System.out.println("Brew Coffee!");
        }
    }

}
